agent/php_zval.h

```diff
@@ -285,17 +285,35 @@ static inline bool nr_php_is_zval_null(const zval* z) {
  *
  * Returns : An ID, or 0 on error.
  *
- * Warning : The type of the given zval is not checked -- if it is not a
- *           resource, you will probably get segfaults!
  */
 static inline long nr_php_zval_resource_id(const zval* zv) {
+  if (!nr_php_is_zval_valid_resource(zv))
+  {
+      return 0;
+  }
 #ifdef PHP7
```

Comment:
Can you please update the style for readability:
`#if ZEND_MODULE_API_NO >= ZEND_7_0_X_API_NO /* PHP 7.0+ */`

Currently PHP7 actually indicates [PHP7+](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_compat.h#L13).